plugins/isplink-client-registration/src/public.php

```diff
@@ -0,0 +1,225 @@
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\GuzzleException;
+
+$configManager = \Ubnt\UcrmPluginSdk\Service\PluginConfigManager::create();
+$config = $configManager->loadConfig();
+$optionsManager = \Ubnt\UcrmPluginSdk\Service\UcrmOptionsManager::create();
+$options = $optionsManager->loadOptions();
+
+// Get UCRM log manager.
+$log = \Ubnt\UcrmPluginSdk\Service\PluginLogManager::create();
+
+$apiError = null;
+$apiSuccess = false;
+
+$publicUrl = str_replace('.php', '/', $options->pluginPublicUrl);
+
+try {
+    $client = new Client([
+        'base_uri' => "api.isplink.net",
+        'headers' => [
+            'Authorization' => 'Bearer ' . $config['API_KEY'],
+            'Content-Type' => 'application/json',
+            'Accept' => 'application/json'
+        ],
+        'timeout' => 5,
+        'http_errors' => true 
+    ]);
+
+    // Test API connection
+    $response = $client->post('/plugin/verify');
+    $apiSuccess = true;
+    $body = (string) $response->getBody();
+    $jsonData = json_decode($body, true);
+    $clientDomain = $jsonData['client_domain'];
+    
+} catch (GuzzleHttp\Exception\ClientException $e) {
+    // Handle 4xx client errors (like 401 Unauthorized)
+    $response = $e->getResponse();
+    $body = (string) $response->getBody();
+    $jsonData = json_decode($body, true);
+    $clientDomain = false;
+    
+    // ISP link is hyperlink to ISP Link website
+    if (isset($jsonData['error']) && $jsonData['error'] === 'Invalid API Key') {
+        $apiError = "An API key and active ISPLink account is required for this plugin. Please contact <a href='https://isplink.app'>ISPLink</a> to activate your account.";
+    } else {
+        $apiError = 'Client error: ' . $response->getStatusCode() . ' - ' . ($jsonData['error'] ?? 'Unknown error');
+    }
+    $log->appendLog('API error: ' . $apiError);
+    
+} catch (GuzzleException $e) {
+    // Handle other Guzzle exceptions (connection issues, timeouts, etc.)
+    $log->appendLog('API error: ' . $e->getMessage());
+}
+
+// If API validation successful, show iframe
+if ($apiSuccess && $clientDomain != false) {
+    ?>
+    <!DOCTYPE html>
+    <html>
+    <head>
+        <title><?php echo htmlspecialchars($config['PAGE_TITLE'] ?? 'ISPLink Client Signup'); ?></title>
+    </head>
+    <body>
+        <iframe 
+            src="https://<?php echo htmlspecialchars($clientDomain); ?>" 
+            style="width: 100%; height: 100vh; border: none;">
+        </iframe>
+    </body>
+    </html>
+    <?php
+    exit;
+}
+
+// Otherwise show the combined form with address check
+?>
+<!DOCTYPE html>
+<html>
+<head>
+    <title><?php echo htmlspecialchars($config['PAGE_TITLE'] ?? 'Sign Up for Service'); ?></title>
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link href="https://cdnjs.cloudflare.com/ajax/libs/bootstrap/5.3.0/css/bootstrap.min.css" rel="stylesheet">
+    <link href="<?php echo $publicUrl . 'styles.css'?>" rel="stylesheet">
```

Comment:
you can just use relative path here, no need for publicUrl
```suggestion
    <link href="public/style.css" rel="stylesheet">
```